source/server/listener_impl.cc

```diff
@@ -1123,6 +1135,30 @@ void ListenerImpl::closeAllSockets() {
   }
 }
 
+bool ListenerMessageUtil::compareSocketOptions(const envoy::config::listener::v3::Listener& lhs,
+                                               const envoy::config::listener::v3::Listener& rhs) {
```

Comment:
Per my other comments in the docs, I think we should NACK the update if reuse_port is false and the options change? Also I think this behavior should be behind a runtime flag.

Also, can we try to have an integration test for both reuse_port true and false in the LDS integration tests for changing socket options?

Reply:
I plan the platform to reject that when binding the listen socket, then we needn't add more code to the envoy, when you update with reuse_port as false, you will get something as below:
```
[2022-09-28 00:55:38.401][65125][error][config] [source/server/listener_manager_impl.cc:1083] listener 'my_listener' failed to bind or apply socket options: cannot bind '0.0.0.0:13333': Address already in use
[2022-09-28 00:55:38.402][65125][warning][config] [source/common/config/filesystem_subscription_impl.cc:60] Filesystem config update rejected: Error adding/updating listener(s) my_listener: cannot bind '0.0.0.0:13333': Address already in use
``

Let me know if we prefer to reject in Envoy instead of platform.

I will try to add integration tests.

thanks!

Reply:
it turns out, we can't test reuse_port as false, since our test is listening on zero port address, so it won't fail due to conflict address when binding port.

Reply:
that makes sense, the message isn't very clear for the user. reject that case should be easy, let me add it. thanks